ghost/activitypub/test/ActivityPubService.test.ts

```diff
@@ -0,0 +1,289 @@
+            knexInstance,
+            siteUrl,
+            logging,
+            identityTokenService as unknown as IdentityTokenService
+        );
+
+        await service.initialiseWebhooks();
+
+        assert(scope.isDone(), 'Expected the ActivityPub site endpoint to be called');
+
+        const webhooks = await knexInstance.select('*').from('webhooks');
+
+        const expectedWebhookCount = 2;
+        const expectedWebhookSecret = 'webhook_secret_baby!!';
+        const expectedWebhookIntegrationId = 'integration_id';
+
+        assert.equal(webhooks.length, expectedWebhookCount);
+
+        for (const webhook of webhooks) {
+            assert.equal(webhook.secret, expectedWebhookSecret);
+            assert.equal(webhook.integration_id, expectedWebhookIntegrationId);
+        }
+
+        await knexInstance.destroy();
+    });
+
+    it('Will not reinitialise webhooks if they are already good', async function () {
+        const knexInstance = await getKnexInstance();
+        await addOwnerUser(knexInstance);
+        await addActivityPubIntegration(knexInstance);
+
+        const siteUrl = new URL('http://fake-site-url');
+        const scope = nock(siteUrl)
+            .get('/.ghost/activitypub/site')
+            .matchHeader('authorization', 'Bearer token:owner@user.com:Owner')
+            .reply(200, {
+                webhook_secret: 'webhook_secret_baby!!'
+            });
+
+        const logging = console;
+        const identityTokenService = {
+            getTokenForUser(email: string, role: string) {
+                return `token:${email}:${role}`;
+            }
+        };
+        const service = new ActivityPubService(
+            knexInstance,
+            siteUrl,
+            logging,
+            identityTokenService as unknown as IdentityTokenService
+        );
+
+        await service.initialiseWebhooks();
+
+        assert(scope.isDone(), 'Expected the ActivityPub site endpoint to be called');
+
+        const webhooks = await knexInstance.select('*').from('webhooks');
+
+        const expectedWebhookCount = 2;
+        const expectedWebhookSecret = 'webhook_secret_baby!!';
+        const expectedWebhookIntegrationId = 'integration_id';
+
+        assert.equal(webhooks.length, expectedWebhookCount);
+
+        for (const webhook of webhooks) {
+            assert.equal(webhook.secret, expectedWebhookSecret);
+            assert.equal(webhook.integration_id, expectedWebhookIntegrationId);
+        }
+
+        nock(siteUrl)
+            .get('/.ghost/activitypub/site')
+            .matchHeader('authorization', 'Bearer token:owner@user.com:Owner')
+            .reply(200, {
+                webhook_secret: 'webhook_secret_baby!!'
+            });
+
+        await service.initialiseWebhooks();
+
+        const webhooksAfterSecondInitialisation = await knexInstance.select('*').from('webhooks');
+
+        assert.deepEqual(webhooksAfterSecondInitialisation, webhooks, 'Expected webhooks to be unchanged');
```

Comment:
To be 100 hundo percent sure, we could check that `knex.del` wasn't called? or do you think this is overkill?